src/script/lua_api/l_object.cpp

```diff
@@ -738,6 +738,28 @@ int ObjectRef::l_get_attach(lua_State *L)
 	return 5;
 }
 
+// get_children(self)
+int ObjectRef::l_get_children(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	ObjectRef *ref = checkobject(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
+
+	//Do it
+	const std::unordered_set<int> child_ids = sao->getAttachmentChildIds();
+	int i = 0;
+	lua_createtable(L, child_ids.size(), 0);
+	for (const int &id : child_ids) {
```

Comment:
even better: 

```suggestion
	for (int i = 0; i < child_ids.size(); i++) {
		ServerActiveObject *child = env->getActiveObject(child_ids[id]);
		getScriptApiBase(L)->objectrefGetOrCreate(L, child);
		lua_rawseti(L, -2, i + 1);
	}
```

Reply:
`++i` if you don't want to rely on the compiler. But really. as long as it works IMO either code is fine.

Reply:
Why? This increases the complexity, worsens readability, adds a bug, for literally no gain

Reply:
This is an unordered set, not a vector.  This doesn't work.

Reply:
Ah, I thought this was a vector. This would be more readable if it were a vector, but won't work with sets

Reply:
ok

> Because the code guidelines say "Don't use iterators when unnecessary."

Actual c++ core guidelines advise for [the opposite](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p3-express-intent), but ok, can't argue with that

Reply:
It's not about the iterator, it's about the `++i` and the initialiation - you already have a traditional for loop there. But this is not applicable as it's a set not a vector

Reply:
I think that part of the guidelines needs some adjustment. (And in practice I haven't seen anybody advocate against iterators recently.)

Since [range-based loops](https://en.cppreference.com/w/cpp/language/range-for) in C++11, iterators are more convenient and *less* bug-prone than simple index loops.
Of course if not convenient, use of iterators should not be forced e.g. if you need the loop index `i` for for a second array.